common/src/main/java/org/jboss/gm/common/utils/FileUtils.java

```diff
@@ -36,19 +36,20 @@ public static String getLastLine(File target) throws IOException {
     }
 
     /**
-     * Take a collection of lines and returns the first non blank entry.
+     * Take a collection of lines and return the first line if one exists. Otherwise, return the empty
+     * string.
      *
      * @param lines the collection of string lines
-     * @return the first non blank entry.
-     * @throws ManipulationException if it cannot find a non-blank entry.
+     * @return the first line if one exists or the empty string otherwise
      */
-    public static String getFirstLine(List<String> lines) throws ManipulationException {
+    public static String getFirstLine(List<String> lines) {
         for (String line : lines) {
             if (StringUtils.isNotBlank(line)) {
                 return line;
             }
         }
-        throw new ManipulationException("Unable to find a non blank line in the collection");
+
+        return StringUtils.EMPTY;
```

Comment:
This change is to allow calling this method on an empty file (which my test creates). I don't quite understand why it just doesn't return

```
lines.isEmpty() ? StringUtils.EMPTY : lines.get(0);
```

but that breaks several of the current tests if done that way.

Reply:
You can't return `lines.get(0)` as you don't know if that is blank or not - you need to iterate through to find the first non-blank one.

Reply:
I'm not convinced by this change ; the file its failing on is `gradle/gme-repos.gradle` which is one that should be overwritten. I don't think we should be changing utility functions to make tests work.

Reply:
Specifically, the change is supposed to allow injecting into an empty file without failing. I did not understand why blank lines were relevant in the first place and why it is not just placed at the very top, blank lines or not.

In any case, I can revert this if I start in my test off with a non-blank script. What is the most minimal script it is looking for, then? As far as the test goes, it was easiest to start with an empty file and then check if the file is not empty as part of the test.

But I wasn't sure it was really a "feature" to be failing in the case of an empty file or file with just whitespace regardless of the test.

Reply:
Reverted.